src/Common/src/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -162,11 +162,14 @@ out instanceByteSizeAndAlignment
             {
                 return ComputeExplicitFieldLayout(type, numInstanceFields);
             }
-            else
+            else if (type.IsSequentialLayout)
```

Comment:
I would just put the condition here to say "is sequential or ProjectN ABI" (type.Context.Target.Abi == ProjectN). ProjectN is the weird one, everything else can do the new logic.

At some point when I get bored I'll port this code to NUTC and we can remove the condition. No need to make things virtual for this because it's temporary.